R/ckanr_settings.R

```diff
@@ -0,0 +1,163 @@
+  cat("  Test behaviour if CKAN offline:", x$test_behaviour)
+}
+
+#------------------------------------------------------------------------------#
+# Setters
+#
+#' Configure default CKAN settings
+#'
+#' @export
+#' @param url A CKAN URL (optional), default: "http://data.techno-science.ca/"
+#' @param key A CKAN API key (optional, character)
+#' @param test_url (optional, character) A valid CKAN URL for testing purposes
+#' @param test_key (optional, character) A valid CKAN API key privileged to
+#'    create datasets at \code{test_url}
+#' @param test_did (optional, character) A valid CKAN dataset ID, existing at
+#'   \code{test_url}
+#' @param test_rid (optional, character) A valid CKAN resource ID, attached to
+#'   \code{did}
+#' @param test_gid (optional, character) A valid CKAN group name at \code{test_url}
+#' @param test_oid (optional, character) A valid CKAN organization name at
+#'   \code{test_url}
+#' @param test_behaviour (optional, character) Whether to fail ("FAIL") or skip
+#' ("SKIP") writing tests in case of problems with the configured test CKAN.
+#' @details
+#' \code{setup_ckanr} sets CKAN connection details. \code{ckanr}'s functions
+#' default to use the default URL and API key unless specified explicitly.
+#'
+#' \code{ckanr}'s automated tests require a valid CKAN URL, a privileged API key
+#' for that URL, plus the IDs of an existing dataset and an existing resource,
+#' repectively.
+#'
+#' The writing tests (create, update, delete) can fail for two reasons:
+#' failures in \code{ckanr}'s code which the tests aim to detect,
+#' or failures in the configured CKAN, which are not necessarily a problem
+#' with \code{ckanr}'s code but prevent the tests to prove otherwise.
+#'
+#' Setting \code{test_behaviour} to \code{"SKIP"} will allow writing tests to skip
+#' if the configured test CKAN fails. This is desirable to e.g. test the other
+#' functions even if the tester has no write access to a CKAN instance.
+#'
+#' Setting \code{test_behaviour} to \code{"FAIL"} will let the tester find any
+#' problems with both the configured test CKAN and the writing functions.
+#'
+#' @examples
+#' # CKAN users without admin/editor privileges could run:
+#' setup_ckanr(url="http://data.techno-science.ca/")
+#'
+#' # Privileged CKAN editor/admin users can run:
+#' setup_ckanr(url="http://data.techno-science.ca/", key="some-CKAN-API-key")
+#'
+#' # ckanR developers/testers can run:
+#' setup_ckanr(url="http://data.techno-science.ca/", key="some-CKAN-API-key",
+#'            test_url="http://test-ckan.gov/",test_key="test-ckan-API-key",
+#'            test_did="test-ckan-dataset-id",test_rid="test-ckan-resource-id",
+#'            test_gid="test-group-name", test_oid="test-organzation-name",
+#'            test_behaviour="FAIL")
+#'
+#' # Not specifying the default CKAN URL will reset the CKAN URL to its default
+#' # "http://data.techno-science.ca/":
+#' setup_ckanr()
+setup_ckanr <- function(
```

Comment:
@florianm I would suggest making this function to set settings the same name format as the function to get settings: So `ckanr_setup` and `ckanr_settings`, that okay?